x/stake/simulation/sim_test.go

```diff
@@ -0,0 +1,369 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+	"github.com/cosmos/cosmos-sdk/x/mock"
+	"github.com/cosmos/cosmos-sdk/x/stake"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+var (
+	stats = make(map[string]int)
+)
+
+// ModuleInvariants runs all invariants of the stake module.
+// Currently: total supply, positive power
+func ModuleInvariants(ck bank.Keeper, k stake.Keeper) mock.Invariant {
+	return func(t *testing.T, app *mock.App, log string) {
+		SupplyInvariants(ck, k)(t, app, log)
+		PositivePowerInvariant(k)(t, app, log)
+		ValidatorSetInvariant(k)(t, app, log)
+	}
+}
+
+// SupplyInvariants checks that the total supply reflects all held loose tokens, bonded tokens, and unbonding delegations
+func SupplyInvariants(ck bank.Keeper, k stake.Keeper) mock.Invariant {
```

Comment:
Maybe this file could be broken out a bit from `sim_test.go` into `Invariants.go`, `msgs.go`, `simulation_test.go`?

Reply:
Done!